diff-containers/test/Test/Data/Map/Diff/Strict.hs

```diff
@@ -43,123 +46,219 @@ import           Test.Util
 -- small values often.
 tests :: TestTree
 tests = testGroup "Data.Map.Diff.Strict" [
-      localOption (QuickCheckTests 1000) $
-      testGroupWithProxy (Proxy @(DiffHistory (OftenSmall Int))) [
+      localOption (QuickCheckTests 10000) $
+      testGroupWithProxy (Proxy @(NormalDiffHistory (OftenSmall Int))) [
           testSemigroupLaws
         , testMonoidLaws
         , testGroupLaws
         ]
     , localOption (QuickCheckTests 1000) $
-      testGroupWithProxy (Proxy @(Diff (OftenSmall Int) (OftenSmall Int))) [
+      testGroupWithProxy (Proxy @(NormalDiff (OftenSmall Int) (OftenSmall Int))) [
           testSemigroupLaws
         , testMonoidLaws
         , testGroupLaws
         ]
-    , localOption (QuickCheckTests 10000) $
-      testProperty "prop_diffingIsPositive" $
-        prop_diffingIsPositive @(OftenSmall Int) @(OftenSmall Int)
-    , localOption (QuickCheckTests 10000) $
-      testProperty "prop_diffThenApply" $
-        prop_diffThenApply @(OftenSmall Int) @(OftenSmall Int)
-    , localOption (QuickCheckTests 10000) $
-      testProperty "prop_applyMempty" $
-        prop_applyMempty @(OftenSmall Int) @(OftenSmall Int)
-    , localOption (QuickCheckMaxRatio 100) $
-      localOption (QuickCheckTests 1000) $
-      testProperty "prop_applyAllAndApplySum" $
-        prop_applyAllAndApplySum @(OftenSmall Int) @(OftenSmall Int)
-    , localOption (QuickCheckMaxRatio 100) $
-      localOption (QuickCheckTests 1000) $
-      testProperty "prop_unsafeApplyAllAndUnsafeApplySum" $
-        prop_unsafeApplyAllAndUnsafeApplySum @(OftenSmall Int) @(OftenSmall Int)
+    , testGroup "diffing" [
+          localOption (QuickCheckTests 10000) $
+          testProperty "prop_diffingIsPositive" $
+            prop_diffingIsPositive @(OftenSmall Int) @(OftenSmall Int)
+        , localOption (QuickCheckTests 10000) $
+          testProperty "prop_diffingIsNormal" $
+            prop_diffingIsNormal @(OftenSmall Int) @(OftenSmall Int)
+        ]
+    , testGroup "Preserving positivity and normality" [
+          localOption (QuickCheckTests 1000) $
```

Comment:
Are we missing a `0` here or is this prop more expensive to test?

Reply:
It's more expensive. I try to keep the running time of most tests within a few seconds